web3/contract.py

```diff
@@ -582,120 +529,386 @@ def _encode_constructor_data(cls, args: Optional[Any] = None,
+            data = add_0x_prefix(
+                encode_abi(self.w3, constructor_abi, arguments, data=self.bytecode)
+            )
+        else:
+            data = to_hex(self.bytecode)
+
+        return data
+
+    @combomethod
+    def _estimate_gas(
+        self, transaction: Optional[TxParams] = None
+    ) -> TxParams:
+        if transaction is None:
+            estimate_gas_transaction: TxParams = {}
+        else:
+            estimate_gas_transaction = cast(TxParams, dict(**transaction))
+            self.check_forbidden_keys_in_transaction(estimate_gas_transaction,
+                                                     ["data", "to"])
+
+        if self.w3.eth.default_account is not empty:
+            # type ignored b/c check prevents an empty default_account
+            estimate_gas_transaction.setdefault('from', self.w3.eth.default_account)  # type: ignore # noqa: E501
+
+        estimate_gas_transaction['data'] = self.data_in_transaction
+
+        return estimate_gas_transaction
+
+    def _get_transaction(self, transaction: Optional[TxParams] = None) -> TxParams:
+        if transaction is None:
+            transact_transaction: TxParams = {}
+        else:
+            transact_transaction = cast(TxParams, dict(**transaction))
+            self.check_forbidden_keys_in_transaction(transact_transaction,
+                                                     ["data", "to"])
+
+        if self.w3.eth.default_account is not empty:
+            # type ignored b/c check prevents an empty default_account
+            transact_transaction.setdefault('from', self.w3.eth.default_account)  # type: ignore
+
+        transact_transaction['data'] = self.data_in_transaction
+
+        return transact_transaction
+
+    @combomethod
+    def _build_transaction(self, transaction: Optional[TxParams] = None) -> TxParams:
+        built_transaction = self._get_transaction(transaction)
+        built_transaction['to'] = Address(b'')
+        return built_transaction
+
+    @staticmethod
+    def check_forbidden_keys_in_transaction(
+        transaction: TxParams, forbidden_keys: Optional[Collection[str]] = None
+    ) -> None:
+        keys_found = transaction.keys() & forbidden_keys
+        if keys_found:
+            raise ValueError(
+                f"Cannot set '{', '.join(keys_found)}' field(s) in transaction"
+            )
+
+
+class ContractConstructor(BaseContractConstructor):
+
+    @combomethod
+    def transact(self, transaction: Optional[TxParams] = None) -> HexBytes:
+        return self.w3.eth.send_transaction(self._get_transaction(transaction))
+
+    @combomethod
+    def build_transaction(self, transaction: Optional[TxParams] = None) -> TxParams:
+        """
+        Build the transaction dictionary without sending
+        """
+        built_transaction = self._build_transaction(transaction)
+        return transactions.fill_transaction_defaults(self.w3, built_transaction)
```

Comment:
We have an existing pattern of importing specific function names from files, so I'd prefer to just import the function directly at the top of the file: 
```
from web3._utils.transactions import (
    fill_transaction_defaults,
)
```

Reply:
done in #2458